src/zeroD/IdealGasConstPressureMoleReactor.cpp

```diff
@@ -0,0 +1,264 @@
+//! @file IdealGasConstPressureMoleReactor.cpp A constant pressure
+//! zero-dimensional reactor with moles as the state
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/zeroD/IdealGasConstPressureMoleReactor.h"
+#include "cantera/zeroD/FlowDevice.h"
+#include "cantera/zeroD/ReactorNet.h"
+#include "cantera/zeroD/ReactorSurface.h"
+#include "cantera/kinetics/Kinetics.h"
+#include "cantera/thermo/ThermoPhase.h"
+#include "cantera/thermo/SurfPhase.h"
+#include "cantera/base/utilities.h"
+#include "float.h"
+
+using namespace std;
+
+namespace Cantera
+{
+
+void IdealGasConstPressureMoleReactor::setThermoMgr(ThermoPhase& thermo)
+{
+    if (thermo.type() != "IdealGas") {
+        throw CanteraError("IdealGasConstPressureMoleReactor::setThermoMgr",
+                           "Incompatible phase type provided");
+    }
+    MoleReactor::setThermoMgr(thermo);
+}
+
+void IdealGasConstPressureMoleReactor::getState(double* y)
+{
+    if (m_thermo == 0) {
+        throw CanteraError("IdealGasConstPressureMoleReactor::getState",
+                           "Error: reactor is empty.");
+    }
+    m_thermo->restoreState(m_state);
+    // get mass for calculations
+    m_mass = m_thermo->density() * m_vol;
```

Comment:
@speth even though I am not using mass as a state variable anymore? I don't do any calculations or updates to mass so I don't see how it would really be up to date.